File: Bot/ExampleCSharp/Game.cs
using Swoq.Interface;

namespace Bot;

public class Game(GameService.GameServiceClient client, StartResponse response)
{
    public string GameId { get; } = response.GameId;
    public int MapWidth { get; } = response.Width;
    public int MapHeight { get; } = response.Height;
    public int VisibilityRange { get; } = response.VisibilityRange;
    public State State { get; private set; } = response.State;

    public void Act(DirectedAction action)
    {
        var request = new ActionRequest() { GameId = GameId, Action = action };

        var response = client.Act(request);
        if (response.Result != Result.Ok)
        {
            throw new GameException($"Act failed (result {response.Result})");
        }

        State = response.State;
    }
}

File: Bot/ExampleGo/game.go
package main

import (
	"bot/proto/swoq"
	"context"
	"fmt"
	"time"
)

type Game struct {
	client          *swoq.GameServiceClient
	gameId          string
	mapWidth        int32
	mapHeight       int32
	visibilityRange int32
	state           *swoq.State
	seed            *int32
	replayFile      *ReplayFile
}

func (g Game) Close() error {
	return nil
}

func (g *Game) Act(action swoq.DirectedAction) error {

	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
	defer cancel()
	actRequest := &swoq.ActRequest{GameId: g.gameId, Action: &action}
	actResponse, err := (*g.client).Act(ctx, actRequest)
	if err != nil {
		return err
	}

	if g.replayFile != nil {
		g.replayFile.Append(actRequest, actResponse)
	}

	if actResponse.Result != swoq.ActResult_ACT_RESULT_OK {
		return fmt.Errorf("Act failed (result %s)", actResponse.Result.String())
	}

	g.state = actResponse.State
	return nil
}
